### PR TITLE
feat(contract): Add sub-message based LS and make LS generic

### PR DESCRIPTION
**Pull request by nabaruns**
1. Add sub-message based LS
2. Make LS generic

### CODE REVIEW

File: src/contract.rs [hunk 1]
```diff
@@ -273,66 +256,50 @@ mod tests {
...
-        let exchange_rate = "0.825537496882794638";
-
-        // Mock each pool in the querier
-        deps.querier
-            .mock_exchange_rate(MOCK_CHAIN_ID.to_string(), exchange_rate.to_string());
+        let msg = Reply {
+            id: 1,
+            result: cosmwasm_std::SubMsgResult::Ok(SubMsgResponse {
+                events: vec![],
+                data: Some(Binary::from_base64("Cv0BeyJjb250cmFjdCI6Indvcm1ob2xlMXl3NHd2MnpxZzl4a242N3p2cTNhenllMHQ4aDB4OWtneWczZDUzanltMjRneHQ0OXZkeXM2czhoN2EiLCJkZW5vbSI6bnVsbCwicmVjaXBpZW50Ijoic2VpMWRrZHdkdmtueDBxYXY1Y3A1a3c2OG1rbjNyOTltM3N2a3lqZnZrenR3aDk3ZHYybG0wa3NqNnhyYWsiLCJhbW91bnQiOiIxMDAwIiwicmVsYXllciI6InNlaTF2aGttMnF2Nzg0cnVseDh5bHJ1MHpwdnl2dzNtM2N5OXgzeHlmdiIsImZlZSI6IjAifQ==").unwrap())
```
Comment (betterclever) [inline src/contract.rs:275]: Why does this base64 data have wormhole and sei chain addresses?
Also, preferably don't put encoded data directly but put json as a raw Rust string and encode it and use it for easier test readability.

**Review by betterclever — COMMENTED** [review]
small comment. rest looks good.

**Comment by betterclever** [comment]
other than that I'm working towards removing the osmosis-std dependency by providing a new persistence-std release. ETA: 1 hour

File: src/contract.rs (hunk 1 above)
Comment (nabaruns) [reply src/contract.rs]: Done.

**Review by betterclever — APPROVED** [review]

**Pull request merged**